src/modules/Dashboard/features/Applet/Activities/Activities.utils.tsx

```diff
@@ -2,45 +2,56 @@ import { t } from 'i18next';
 
 import { Svg } from 'shared/components/Svg';
 import { MenuItem, MenuItemType } from 'shared/components';
+import { isManagerOrOwner } from 'shared/utils';
+import { Roles } from 'shared/consts';
 
 import { ActivityActions, ActivityActionProps } from './Activities.types';
 
 export const getActivityActions = ({
   actions: { editActivity, exportData, assignActivity, takeNow },
   appletId,
   activityId,
-}: ActivityActions): MenuItem<ActivityActionProps>[] => [
-  {
-    icon: <Svg id="edit" />,
-    action: editActivity,
-    title: t('editActivity'),
-    context: { appletId, activityId },
-    isDisplayed: true,
-    'data-testid': 'dashboard-applets-activities-activity-edit',
-  },
-  {
-    icon: <Svg id="export" />,
-    action: exportData,
-    title: t('exportData'),
-    context: { appletId, activityId },
-    isDisplayed: true,
-    'data-testid': 'dashboard-applets-activities-activity-export',
-  },
-  { type: MenuItemType.Divider },
-  {
-    icon: <Svg id="add" />,
-    action: assignActivity,
-    title: t('assignActivity'),
-    context: { appletId, activityId },
-    isDisplayed: true,
-    'data-testid': 'dashboard-applets-activities-activity-assign',
-  },
-  {
-    icon: <Svg id="play-outline" />,
-    action: takeNow,
-    title: t('takeNow'),
-    context: { appletId, activityId },
-    isDisplayed: true,
-    'data-testid': 'dashboard-applets-activities-activity-take-now',
-  },
-];
+  dataTestid,
+  roles,
+}: ActivityActions): MenuItem<ActivityActionProps>[] => {
+  const canEdit =
```

Comment:
I suppose it will be easy to move this logic into a `checkIfCanEdit` (for ex.) function and tests it separately, does it make sense?

Reply:
Yes that makes sense! I'll move it out and add tests

Reply:
Done!